### PR TITLE
Feature/us6025 call number persistence

### PR DESCRIPTION
**Pull request by dtkocher**
**Rally**: https://rally1.rallydev.com/#/27593764268d/detail/userstory/80805666784/tasks

**Build**: http://mp-ci.centralus.cloudapp.azure.com/viewLog.html?buildId=13545&tab=buildResultsDiv&buildTypeId=Code_Development_SignInCheckIn_Feature

### CODE REVIEW

File: SignInCheckIn/MinistryPlatform.Translation/Repositories/ParticipantRepository.cs [hunk 1]
```diff
@@ -100,5 +100,37 @@ public List<MpGroupParticipantDto> CreateGroupParticipants(string token, List<Mp
 
             return _ministryPlatformRestRepository.UsingAuthenticationToken(token).Create(mpGroupParticipantDtos, groupParticipantColumns);
         }
+
```
Comment (dtkocher) [inline SignInCheckIn/MinistryPlatform.Translation/Repositories/ParticipantRepository.cs]: Not sure why this functions is needed.  I was thinking we would just set the call number for the second event when we create the participant record.

File: SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs [hunk 2]
```diff
@@ -132,9 +132,13 @@ public ParticipantEventMapDto SigninParticipants(ParticipantEventMapDto particip
                     {
                         subItem.AssignedSecondaryRoomId = item.AssignedRoomId;
                         subItem.AssignedSecondaryRoomName = item.AssignedRoomName;
+                        subItem.CallNumber = item.CallNumber;
```
Comment (dtkocher) [inline SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs]: This should only be the last 4 of the Event Participant ID currently it is null.

File: SignInCheckIn/MinistryPlatform.Translation/Repositories/Interfaces/IMinistryPlatformRestRepository.cs [hunk 3]
```diff
@@ -31,6 +31,8 @@ public interface IMinistryPlatformRestRepository
 
         T Update<T>(T objectToUpdate, List<string> selectColumns);
         T Update<T>(T objectToUpdate, string selectColumns = null);
+        List<T> Update<T>(List<T> objectToUpdate, List<string> selectColumns);
```
Comment (andycanterbury) [inline SignInCheckIn/MinistryPlatform.Translation/Repositories/Interfaces/IMinistryPlatformRestRepository.cs]: Your implementation calls the first param object**s**ToUpdate. Should probably be consistent, and plural makes more sense with the list.

File: SignInCheckIn/MinistryPlatform.Translation/Repositories/Interfaces/IMinistryPlatformRestRepository.cs [hunk 4]
```diff
@@ -31,6 +31,8 @@ public interface IMinistryPlatformRestRepository
...
+        List<T> Update<T>(List<T> objectToUpdate, string selectColumns = null);
```
Comment (andycanterbury) [inline SignInCheckIn/MinistryPlatform.Translation/Repositories/Interfaces/IMinistryPlatformRestRepository.cs]: Your implementation calls the first param objectsToUpdate. Should probably be consistent, and plural makes more sense with the list.

File: SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs [hunk 5]
```diff
@@ -123,29 +123,68 @@ public ParticipantEventMapDto SigninParticipants(ParticipantEventMapDto particip
...
+            // Add back those participants that didn't get a room assigned
+             response.Participants.AddRange(participantEventMapDto.Participants.Where(p => !p.AssignedRoomId.HasValue && p.Selected));
+            response.Participants.ForEach(p => p.Selected = true);
+
+            return response;
+        }
+
+        private void SetParticipantsPrintInformation(List<ParticipantDto> participants, IReadOnlyList<MpEventDto> eventsForSignin)
+        {
+            if (eventsForSignin.Count == 1)
```
Comment (andycanterbury) [inline SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs]: It looks like previously the code was checking for a count of 2. Was the change to 1 intentional?

Comment (dtkocher) [inline SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs]: I split this into 3 function calls one that handles if count is 1 (one event) and one that handles if count is 2 (two events)

**Pull request merged**